### ISSUE TITLE
Downloading (-1 peers)

### ISSUE THREAD

**Issue by hisener** [wmhilton/download-with-webtorrent-button#1]
I've started the downloading on the demo page and after some time, the result is:

![screenshot from 2017-04-17 10-41-45](https://cloud.githubusercontent.com/assets/7688949/25082624/adf680f6-235a-11e7-8a67-169d75967fe3.png)

And I could not download the file.

**Comment by billiegoose** [comment]
What the frick. It was totally working on Friday. I'm getting this error too! I'm going to split this into two issues though: one for the -1 peers and one for the download not working.

@feross I'm subtracting 1 from `numPeers`, because I thought `numPeers` was counting itself in the total. Is that incorrect? Or does it count itself while its downloading and then leave once the download is complete or something? Is there a way I can subdivide that total into the number of peers I'm actively downloading from and uploading to?

**Comment by feross** [comment]
Web seeds are counted in `numPeers`. Perhaps that's why you were always seeing at least 1. But, it's an artifact of our implementation that we treat web seeds as normal peers. So sometimes they can get disconnected. I wouldn't subtract 1 from the total number, personally.

**Comment by billiegoose** [comment]
That... actually explains all the behavior I've seen. I don't subtract 1 anymore.
